app/controllers/admin/admin_users_controller.rb

```diff
@@ -39,76 +39,115 @@ def update_user
+      elsif params[:next_of_kin_name].blank? && params[:next_of_kin_email].present?
+        flash[:error] = ts('Fannish next of kin user is missing.')
+        redirect_to request.referer || root_path
+
+      # there is a next of kin username, but it is not a valid user
+      elsif params[:next_of_kin_name].present? && submitted_kin_user.nil?
+        flash[:error] = ts('Fannish next of kin user is invalid.')
+        redirect_to request.referer || root_path
+
+      # there is an admin action selected, but no note entered
+      elsif params[:admin_action].present? && @admin_note.blank?
+        flash[:error] = ts('You must include notes in order to perform this action.')
+        redirect_to request.referer || root_path
+
+      # there is no length entered for the suspension
+      elsif params[:admin_action] == 'suspend' && params[:suspend_days].blank?
+        flash[:error] = ts('Please enter the number of days for which the user should be suspended.')
+        redirect_to request.referer || root_path
+
+      # we made it through without any errors, so let's do some stuff
       else
-        if params[:admin_action] == 'warn'
-          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_WARN, :note => @admin_note, :admin_id => current_admin.id})
-          flash[:notice] = ts("Warning was recorded")
-          redirect_to(request.env["HTTP_REFERER"] || root_path)
-        elsif params[:admin_action] == 'suspend'
-          if params[:suspend_days].blank?
-            flash[:error] = ts("Please enter the number of days for which the user should be suspended.")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
+        success_message = []
+
+        # find or create a fannish next of kin if the fields are filled in
+        if params[:next_of_kin_name].present? && params[:next_of_kin_email].present?
+          if @user.fannish_next_of_kin.nil?
+            @user.fannish_next_of_kin = FannishNextOfKin.new(user_id: params[:user_login],
+              kin_id: submitted_kin_user.id,
+              kin_email: params[:next_of_kin_email])
+            success_message << ts('Fannish next of kin added.')
           else
-            @user.suspended = true
-            @suspension_days = params[:suspend_days].to_i
-            @user.suspended_until = @suspension_days.days.from_now
-            if @user.save && @user.suspended? && !@user.suspended_until.blank?
-              @user.create_log_item( options = {:action => ArchiveConfig.ACTION_SUSPEND, :note => @admin_note, :admin_id => current_admin.id, :enddate => @user.suspended_until})
-              flash[:notice] = ts("User has been temporarily suspended")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            else
-              flash[:error] = ts("User could not be suspended")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            end
+            @user.fannish_next_of_kin = @user.fannish_next_of_kin
```

Comment:
You're right! I could swear I had a reason for doing that, but I can no longer find it.